iotdb-client/client-py/iotdb/SessionPool.py

```diff
@@ -0,0 +1,122 @@
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+import multiprocessing
+import time
+from queue import Queue
+from threading import Lock
+
+from iotdb.Session import Session
+
+DEFAULT_MULTIPIE = 5
+DEFAULT_FETCH_SIZE = 5000
+DEFAULT_MAX_RETRY = 3
+DEFAULT_TIME_ZONE = "Asia/Shanghai"
+
+
+class PoolConfig(object):
+    def __init__(self, host: str, ip: str, user_name: str, password: str, node_urls: list = None,
+                 fetch_size: int = DEFAULT_FETCH_SIZE, time_zone: str = DEFAULT_TIME_ZONE,
+                 max_retry: int = DEFAULT_MAX_RETRY):
+        self.host = host
+        self.ip = ip
+        if node_urls is None:
+            node_urls = []
+        self.node_urls = node_urls
+        self.user_name = user_name
+        self.password = password
+        self.fetch_size = fetch_size
+        self.time_zone = time_zone
+        self.max_retry = max_retry
+
+
+class SessionPool(object):
+
+    def __init__(self, pool_config: PoolConfig, max_pool_size: int, wait_timeout_in_ms: int):
+        self.__pool_config = pool_config
+        self.__max_pool_size = max_pool_size
+        self.__wait_timeout_in_ms = wait_timeout_in_ms / 1000
+        self.__pool_size = 0
+        self.__queue = Queue(max_pool_size)
+        self.__lock = Lock()
+
+    def __construct_session(self) -> Session:
+        if len(self.__pool_config.node_urls) > 0:
+            return Session.init_from_node_urls(self.__pool_config.node_urls, self.__pool_config.user_name,
+                                               self.__pool_config.password, self.__pool_config.fetch_size,
+                                               self.__pool_config.time_zone)
+
+        else:
+            return Session(self.__pool_config.host, self.__pool_config.ip, self.__pool_config.user_name,
+                           self.__pool_config.password, self.__pool_config.fetch_size, self.__pool_config.time_zone)
+
+    def __poll_session(self) -> Session | None:
+        if self.__queue.empty():
+            return None
+        return self.__queue.get(block=False)
+
+    def get_session(self) -> Session:
+        should_create = False
+        start = time.time()
+
+        session = self.__poll_session()
+        while session is None:
+            self.__lock.acquire()
+            if self.__pool_size < self.__max_pool_size:
+                self.__pool_size += 1
+                should_create = True
+                self.__lock.release()
+                break
+            else:
+                if time.time() - start > self.__wait_timeout_in_ms:
+                    self.__lock.release()
+                    raise TimeoutError("Wait to get session timeout in SessionPool, current pool size: {0}"
+                                       .format(self.__max_pool_size))
+                time.sleep(1)
```

Comment:
sleep for 1 second? Make sure default time unit in python sleep is second.

Reply:
Yes, the unit of time.sleep() is second. 
https://docs.python.org/3/library/time.html?highlight=sleep#time.sleep